lms/envs/common.py

```diff
@@ -1298,6 +1298,12 @@ def _make_locale_paths(settings):
 
     'edx_rest_framework_extensions.middleware.EnsureJWTAuthSettingsMiddleware',
 
+    # Clears cache as appropriate.
+    'edx_django_utils.cache.middleware.CacheUtilsMiddleware',
```

Comment:
@ormsbee: Most of the following questions are about the middleware itself, but I'm documenting here so we can resolve (if needed) before landing this.

1. In ecommerce, we had some unexplained cache middleware spikes for the middleware we moved.  See https://openedx.atlassian.net/browse/EG-5
2. Nimisha thought the Request Cache shouldn't need to be reset if using a thread local.  Is the CacheUtilsMiddleware semi-pointless?
3. Should the middleware be renamed given the app is no longer called cache_utils?
4. Where should we document the middleware dependencies, if there still is one for RequestMetricsMiddleware on CacheUtilsMiddleware?

FYI: @douglashall

Reply:
1. I don't know anything about the ecommerce issues, but I think we've run into a number of odd corner cases that pop up because it uses gevent -- perhaps this is related as well?
2. I haven't mucked with thread internals of Django, but my understanding is that threads are reused across requests in Django instead of being spun up as new requests come in, and that therefore cleanup is necessary.
3. A more descriptive name would be better. Even before the rename, `CacheUtilsMiddleware` was already kinda fuzzy. Maybe `RequestCacheMiddleware`?
4. So for most runtime setup dependencies, you'd probably do a system check in the app config's `ready()` method, but I don't think there's a place you can plug into that for middleware. You can define an `__init__` on a middleware class though, so I think that's the layer you'd want to check-and-explode if the dependency is not present.